examples/summarization/t5/evaluate_cnn.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+from pathlib import Path
+
+from tqdm import tqdm
+
+from transformers import T5Tokenizer, TFT5ForConditionalGeneration
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def generate_summaries(lns, out_file, batch_size):
+    fout = Path(out_file).open("w")
+
+    model = TFT5ForConditionalGeneration.from_pretrained("t5-large")
+    tokenizer = T5Tokenizer.from_pretrained("t5-large")
+
+    max_length = 140
+    min_length = 55
+
+    for batch in tqdm(list(chunks(lns, batch_size))):
+        batch = ["summarize:" + text for text in batch]
+
+        dct = tokenizer.batch_encode_plus(batch, max_length=512, return_tensors="tf", pad_to_max_length=True)
+        summaries = model.generate(
+            input_ids=dct["input_ids"],
+            attention_mask=dct["attention_mask"],
+            num_beams=4,
+            length_penalty=2.0,
+            max_length=max_length,
+            min_length=min_length,
+            early_stopping=True,
+            bos_token_id=tokenizer.pad_token_id,
```

Comment:
@craffel - I use these params as default params for t5 summarization. I saw in the paper that you use `num_beams=4` as well and from the code: 

I assumed that you pad to `512`. `transformers` uses a different `length_penalty` so instead of `alpha=0.6`, I set it here to `2.0` - do you think that is good? It's the default parameter for Bart. And regarding min_length and max_length, I am not sure whether these params are good - they are copied form the Bart defaults.

Reply:
Hey Patrick, thanks for making this. Some notes
- The released checkpoints are after multitask pre-training. You will get decent summarization results with them, but in order to get our SoTA-at-the-time summarization results you'll need to finetune the model further before generating from it. You can probably ignore that for the sake of an example, but it may be worth mentioning in the README.
- We didn't play with the num beams or length penalty at all, just set them to values that had been used in the past on machine translation and never tweaked them at all. So, your values are probably fine (especially since you already are going to be losing some performance without fine-tuning).
- We don't have min length/max length functionality in our code; we just let the model choose whatever length. If BART uses this trick, it probably helps, and it seems somewhat model-agnostic so we can just use their values.
- The model should work equivalently whether you pad to a max length or not. You should also be able to use a sequence length longer than 512 and it should work fine too.

Reply:
Thanks a lot!